main/systemd.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+    "fmt"
+    "os"
+    "os/exec"
+    "strings"
+    "path/filepath"
+)
+
+func contains(distros []string, distro string) bool {
+    if distro != "" {
+        for _, v := range distros {
+            if v == distro {
+                return true
+            }
+        }
+    }
+    return false
+}
+
+func get_distro() string {
+    path, _ := os.Getwd()
+    output, err := exec.Command("sudo", filepath.Join(path, "scripts/distro_script.sh")).Output()
+    if err != nil {
+        errorMessage := fmt.Sprintf("Error getting distro name: %+v", err)
+		errorLogger.Println(errorMessage)
+    }
+    distroName := strings.ToLower(string(output))
```

Comment:
If the previous call fails, the value of `output` would be incorrect, we should not try to derive the distro name in that case and fail the function

Reply:
I tested it, you get empty output and won't break the function. At the end of it we get default systemdpath if distro not found.

Reply:
wrong place for handling this functionality. This default behavior should be one level up, not in this utility function

Reply:
Actually, default behavior was in another function not here. I have changed now as part of error handling.

Reply:
Would `err` not be non-nil in the cause of the previous call failing? I think this is covered by the early return.

Reply:
yeah, err would be non-nil in case of failure and returning error which I have addressed now as part of error handling